src/Lib.P2P/Discovery/MdnsJs.cs

```diff
@@ -0,0 +1,108 @@
+using Makaretu.Dns;
+using MultiFormats;
+
+namespace Lib.P2P.Discovery
+{
+    /// <summary>
+    ///   Discovers peers using Multicast DNS according to
+    ///   js-ipfs v0.32.3
+    /// </summary>
+    public class MdnsJs : Mdns
+    {
+        /// <summary>
+        ///   Creates a new instance of the class.  Sets the <see cref="Mdns.ServiceName"/>
+        ///   to "ipfs".
+        /// </summary>
+        public MdnsJs() { ServiceName = "ipfs"; }
+
+        /// <inheritdoc />
+        protected override void OnServiceDiscovery(ServiceDiscovery discovery)
+        {
+            discovery.AnswersContainsAdditionalRecords = true;
+        }
+
+        /// <inheritdoc />
+        public override ServiceProfile BuildProfile()
+        {
+            // Only internet addresses.
+            var addresses = LocalPeer.Addresses
+               .Where(a => a.Protocols.First().Name == "ip4" || a.Protocols.First().Name == "ip6")
+               .ToArray();
+            if (addresses.Length == 0) return null;
+            var ipAddresses = addresses
+               .Select(a => IPAddress.Parse(a.Protocols.First().Value));
+
+            // Only one port is supported.
+            var tcpPort = addresses.First()
+               .Protocols.First(p => p.Name == "tcp")
+               .Value;
+
+            // Create the DNS records for this peer.  The TXT record
+            // is singular and must contain the peer ID.
+            var profile = new ServiceProfile(
+                LocalPeer.Id.ToBase58(),
+                ServiceName,
+                ushort.Parse(tcpPort),
+                ipAddresses
+            );
+            profile.Resources.RemoveAll(r => r.Type == DnsType.TXT);
+            var txt = new TXTRecord {Name = profile.FullyQualifiedName};
+            txt.Strings.Add(profile.InstanceName.ToString());
+            profile.Resources.Add(txt);
+
+            return profile;
+        }
+
+        /// <inheritdoc />
+        public override IEnumerable<MultiAddress> GetAddresses(Message message)
+        {
+            var qsn = ServiceName + ".local";
+            var peerNames = message.Answers
+               .OfType<PTRRecord>()
+               .Where(a => a.Name == qsn)
+               .Select(a => a.DomainName);
+            foreach (var name in peerNames)
+            {
+                var id = name.Labels[0];
+                var srv = message.Answers
+                   .OfType<SRVRecord>()
+                   .First(r => r.Name == name);
+                var aRecords = message.Answers
+                   .OfType<ARecord>()
+                   .Where(a => a.Name == name || a.Name == srv.Target);
+                foreach (var a in aRecords) yield return new MultiAddress($"/ip4/{a.Address}/tcp/{srv.Port}/ipfs/{id}");
+                var aaaaRecords = message.Answers
+                   .OfType<AAAARecord>()
+                   .Where(a => a.Name == name || a.Name == srv.Target);
+                foreach (var a in aaaaRecords)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'foreach' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)